modules/ROOT/pages/cloud-deployments/neo4j-cluster-cloud.adoc

```diff
@@ -0,0 +1,109 @@
+
+For information on how to manage databases and servers in a cluster, see respectively xref::clustering/databases.adoc[] and xref::clustering/servers.adoc[].
+
+
+== Neo4j cluster on AWS
+
+Neo4j does not provide Amazon Machine Images (AMIs) with a pre-installed version of the product.
+The Neo4j AWS Marketplace listings (and listings on GitHub) use CloudFormation templates that deploy and configure Neo4j dynamically with a shell script.
+
+
+// === Neo4j cluster and auto-scaling groups on AWS
+
+
+=== Removing a (secondary constrained) server from the cluster
+
+Imagine you have a cluster consisting of three primary constrained servers and two secondary constrained servers.
+This means that three servers host primary databases and the other two host secondary databases.
+
+When performing rolling updates on Amazon Machine Images (AMIs) for secondary servers, it is important to follow a structured approach.
+Rotating AMIs is a common practice in such environments.
+
+However, simply removing secondary servers from the target Network Load Balancer (NLB) one by one does not prevent read requests from being routed to them.
+This occurs because the NLB and Neo4j server-side routing operate independently and do not share awareness of server availability.
+
+To correctly remove a secondary server from the cluster and reintroduce it after the update:
+
+. Remove the server from the NLB to stop traffic routing.
+. Shut down the server before proceeding with the AMI update.
+
+
+Here are the steps:
+
+. Remove the secondary from the AWS NLB.
+ This prevents external clients from sending requests to the secondary.
+
+. Since Neo4j's cluster routing (server-side routing) does not use the NLB, you need to ensure that queries are not routed to the secondary server.
+To do this, you have to cleanly shut down the secondary.
+
+.. Run the following query to check servers are hosting all their assigned databases.
+The query should return no results:
++
+[source, cypher, role=noplay]
+----
+SHOW SERVERS YIELD name, hosting, requestedHosting, serverId WHERE requestedHosting <> hosting
+----
+
+.. Use the following query to check all databases are in their expected state.
+The query should return no results:
++
+[source, cypher, role=noplay]
+----
+SHOW DATABASES YIELD name, address, currentStatus, requestedStatus, statusMessage WHERE currentStatus <> requestedStatus RETURN name, address, currentStatus, requestedStatus, statusMessage
+----
+
+.. To stop the Neo4j service, run the following command:
++
+[source, shell, role=copy]
+----
+sudo systemctl stop neo4j
+----
++
+To configure the timeout period for waiting on active transactions to either complete or be terminated during shutdown, you can modify the environment variable `NEO4J_SHUTDOWN_TIMEOUT` using `systemctl edit neo4j.service`
+or the setting xref::configuration/configuration-settings.adoc#config_db.shutdown_transaction_end_timeout[`db.shutdown_transaction_end_timeout`] in _neo4j.conf_ file.
```

Comment:
These two values do different things. `NEO4J_SHUTDOWN_TIMEOUT` is used by the service wrapper (`sysctl` script here) to decide how long to wait for the process to exit on its own before killing it. `db.shutdown_transaction_end_timeout` is used within the Neo4j process to decide how long to wait for a running transaction to end when shutting down the database, before just terminating the _transaction_ and letting database shutdown finish.